net/https-dns-proxy/files/README.md

```diff
@@ -1,94 +1,3 @@
-
-config https-dns-proxy
-  option bootstrap_dns '8.8.8.8,8.8.4.4'
-  option resolver_url 'https://dns.google/dns-query'
-  option listen_addr '127.0.0.1'
-  option listen_port '5053'
-  option user 'nobody'
-  option group 'nogroup'
-
-config https-dns-proxy
-  option bootstrap_dns '1.1.1.1,1.0.0.1'
-  option resolver_url 'https://cloudflare-dns.com/dns-query'
-  option listen_addr '127.0.0.1'
-  option listen_port '5054'
-  option user 'nobody'
-  option group 'nogroup'
-```
-
-The ```update_dnsmasq_config``` option can be set to dash (set to ```'-'``` to not change ```DNSMASQ``` server settings on start/stop), can be set to ```'*'``` to affect all ```DNSMASQ``` instance server settings or have a space-separated list of ```DNSMASQ``` instances to affect (like ```'0 4 5'```). If this option is omitted, the default setting is ```'*'```.
-
-Starting with ```https-dns-proxy``` version ```2019-12-03-3``` and higher, when the service is set to update the DNSMASQ servers setting on start/stop, it does not override entries which contain either ```#``` or ```/```, so the entries like listed below will be kept in use:
-
-```test
-  list server '/onion/127.0.0.1#65453'
-  list server '/openwrt.org/8.8.8.8'
-  list server '/pool.ntp.org/8.8.8.8'
-  list server '127.0.0.1#15353'
-  list server '127.0.0.1#55353'
-  list server '127.0.0.1#65353'
-```
-
-The https-dns-proxy instance settings are:
-
-|Parameter|Type|Default|Description|
-| --- | --- | --- | --- |
-|bootstrap_dns|IP Address||The non-encrypted DNS servers to be used to resolve the DoH server name on start.|
-|listen_addr|IP Address|127.0.0.1|The local IP address to listen to requests.|
-|listen_port|port|5053 and up|If this setting is omitted, the service will start the first https-dns-proxy instance on port 5053, second on 5054 and so on.|
-|logfile|Full filepath||Full filepath to the file to log the instance events to.|
-|resolver_url|URL||The https URL to the RFC8484-compatible resolver.|
-|proxy_server|URL||Local proxy server to use when accessing resolvers.|
-|user|String|nobody|Local user to run instance under.|
-|group|String|nogroup|Local group to run instance under.|
-|use_http1|Boolean|0|If set to 1, use HTTP/1 on installations with broken/outdated ```curl``` package. Included for posterity reasons, you will most likely not ever need it on OpenWrt.|
-|verbosity|Integer|0|logging verbosity level. fatal = 0, error = 1, warning = 2, info = 3, debug = 4|
-|use_ipv6_resolvers_only|Boolean|0|If set to 1, Forces IPv6 DNS resolvers instead of IPv4|
-
-## Thanks
-
-This OpenWrt package wouldn't have been possible without [@aarond10](https://github.com/aarond10)'s [https-dns-proxy](https://github.com/aarond10/https_dns_proxy) and his active participation in the OpenWrt package itself. Special thanks to [@jow-](https://github.com/jow-) for general package/luci guidance.
+README has been moved to [https://docs.openwrt.melmac.net/https-dns-proxy/](https://docs.openwrt.melmac.net/https-dns-proxy/).
```

Comment:
I'd prefer you leave the README here (and format the table nicely via e.g. [remark](https://github.com/remarkjs/remark)), lose the ~tracker~ ~counter~ badge and use the [*official* stats](https://downloads.openwrt.org/stats/awstats.downloads.openwrt.org.allextra2.html).

I'll ask the download maintainers if they can beautify the stats a bit, ideally merging stats for single packages, instead of (just) showing it per architecture, which makes a *total downloads* count a bit tedious. Seems like you got multiple hundreds of installs :rocket:. 

I'll also ask about the JavaScript snipped you posted, from my understanding it is not really required to track which files are downloaded or Wiki pages are visited.

If you wanted the counter to increase *visibility* of your project, we could think of a better way like *featured LuCI apps* via blog posts or forum entries.

Reply:
> I'd prefer you leave the README here (and format the table nicely via e.g. [remark](https://github.com/remarkjs/remark)), lose the ~tracker~ ~counter~ badge and use the [_official_ stats](https://downloads.openwrt.org/stats/awstats.downloads.openwrt.org.allextra2.html).

Could you please elaborate on "formatting the table nicely"? 

For keeping README here -- what's the plan for screenshots? I would strongly prefer to keep the screenshots hosted together with the README (not an option for the source code repo as far as I understand) and, having experienced it now, I would strongly prefer an option to have the README published thru GitHub pages (or some such, after the move), so it opens full height/width, without GitHub header/side navigation. 

> I'll also ask about the JavaScript snipped you posted, from my understanding it is not really required to track which files are downloaded or Wiki pages are visited.

As far as I remember the awstats is a web-site log analyzer, but back when I used it, they didn't have the javascript helper, so I'm puzzled what's the reason to use one now. It's simply weird that if there's a crusade against hits counter, there's tracker javascript code on the home page which no one has objected to.

Reply:
> Could you please elaborate on "formatting the table nicely"?

If you scroll down a bit you see a "nicely formatted" table in the [contributing document](https://raw.githubusercontent.com/openwrt/packages/master/CONTRIBUTING.md)

> For keeping README here -- what's the plan for screenshots? I would strongly prefer to keep the screenshots hosted together with the README (not an option for the source code repo as far as I understand) and, having experienced it now, I would strongly prefer an option to have the README published thru GitHub pages (or some such, after the move), so it opens full height/width, without GitHub header/side navigation.

I think that's a fair point, screenshots within `packages.git` make no sense. The discussion about the counter seems a waste of time to me. Hosting anything on GitHub involves much more tracking than just a hit counter, so if anyone is concerned about luring people into being tracked, we shouldn't offer `packages.git` on GitHub at all. On the other hand what do you get from this counter? How many people read your documentation? You could host it yourself and check Nginx access logs. Or, as stated before, you look at the official stats and actually get a idea of how many people actually install the package (minus all the mirror pulls).

> I'll also ask about the JavaScript snipped you posted, from my understanding it is not really required to track which files are downloaded or Wiki pages are visited.

I think it's mostly legacy which where added in good intentions some long time ago. I pinged Thomas and kindly asked for removal.

Reply:
Thanks, installed remark processor and reformatted all the READMEs.

Hits on READMEs approximate how many people install packages from my own repo (where I usually post test versions before submitting PRs). Before I switched to a docs repo there were thousands on README hits for vpn-policy-routing for example.

For the issue at hand -- until there's a solution to host and publish READMEs and the screenshots on OpenWrt infrastructure, what's the proposed solution for now?

Reply:
I think it's cleaner to have only a single README.md in your own repo. If you feel like adding a counter, that's your choice - I usually block/delete cookies from such sites, if other people happen to do the same your results aren't very accurate, but that's up to you.

@neheb @BKPepe @hnyman any objections?

Reply:
@aparcar I feel that it's been enough time for anyone to voice any objections. If there's no separate OpenWrt repo for READMEs which can also host screenshots, I'd prefer to keep READMEs and screenshots together at mine. Please advise if there are any other issues preventing this (and https://github.com/openwrt/packages/pull/13513 and https://github.com/openwrt/packages/pull/13514) from merging. Thanks!